src/main/java/pl/touk/sputnik/connector/gerrit/JGitCommentFilter.java

```diff
@@ -0,0 +1,101 @@
+package pl.touk.sputnik.connector.gerrit;
+
+import static java.lang.Integer.parseInt;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+
+@Slf4j
+public class JGitCommentFilter implements CommentFilter {

```

Comment:
I understand you intensions, which is great, but I don't like implementation.

First case is jgit itself. This is a second dependency for git and it may use different channel or protocol to connect git. Two dependencies may be overhaul. I'd like to stick to Gerrit REST API. It has Get Diff operation that provides detailed information. 

Parsing diff does not belong to Sputnik I guess. It should take place in GerritApi and return dedicated object for diff result - your FileDiff is a good start. You should create pull request for GerritApi library, @uwolfer would love that.

Also, your implementation lacks unit tests. Long method may be error-prone.